### DISCUSSION TITLE
Bintray will no longer be available after 01-May-2021

### DISCUSSION THREAD

**Discussion by bdeshi** [OpenBangla/OpenBangla-Keyboard#191] [General]
https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/
> _To streamline the productivity of the JFrog Platform we will be sunsetting Bintray (including JCenter), GoCenter, and ChartCenter services on May 1st, 2021. Users of these services will need to migrate to the respective canonical repository_

possible alternatives:
- [ ] selfhosted
- [ ] github
- [ ] packagecloud
- [ ] openSUSE Build Service
- [ ] include in official distro repos
- [ ] ...

**Comment by bdeshi** [comment]
@mominul @Adyel @ahmubashshir

**Reply by Adyel** [reply]
I don't like depending on 3rd party services because you can never know when they will change their mind. I think going with GitHub is the best solution. We have to maintain all the install script (like we used to) but I think this is better than hosting it and messing with PPA and whatnot.

**Reply by bdeshi** [reply]
to be honest, I'm not a fan of the install.sh script. I think the most user-friendly distribution methods are either bare dev/rpm packages as [releases](https://github.com/OpenBangla/OpenBangla-Keyboard/releases) or package repositories as we have in bintray, instead of tacking on more and more distro detection into install.sh, as we used to.
Since the source code is hosted on github already, we can look into hosting package repos here as well, this minimises dependency on 3rd party services.

**Reply by mominul** [reply]
I don't like the install.sh script way too. The ideal thing would be getting OBK into the respective official package repositories. But the code and build system is quite messy from my perspective and I want to land a major change to OBK in the next version, so I am quite unsure about getting the attention of the repo maintainers in this state. Furthermore, I don't know about their packaging policies either.

> Since the source code is hosted on github already, we can look into hosting package repos here as well, this minimises dependency on 3rd party services.

I am not aware of the fact the Github features hosting package repos such as for deb, rpm, etc, did they just introduced this feature? It'd be great if it was.

**Reply by ahmubashshir** [reply]
can't we host it as git-lfs?

**Reply by bdeshi** [reply]
> I am not aware of the fact the Github features hosting package repos such as for deb, rpm, etc, did they just introduced this feature? It'd be great if it was.

@mominul github doesn't officially have support for distro package repostories, but a public github repo may be [ab]used for this. I'm planning to check and prepare a demo... somewhen... by may.

@ahmubashshir github places a 1gb monthly lfs data download limit on free github accounts.

**Reply by ahmubashshir** [reply]
@bdeshi can't we use [Open Build Service](https://build.opensuse.org/) by OpenSUSE?

**Reply by bdeshi** [reply]
@ahmubashshir cool find! looks promising.

**Comment by mominul** [comment]
@ahmubashshir Great find!

Lately, I am thinking of trying OBK to get into the official distro package repositories. I think #183 has some issues we need to fix to prepare OBK for inclusion into the official repos. I am thinking of releasing a new minor version with all of the required fixes we'll need to get OBK into the official repos (Debian, Fedora, etc.).
Is anyone here with any previous experience of getting in touch with the Debian or fedora package maintainer?

@bdeshi We had two AUR packages for OBK, so the situation for Arch Linux will be good enough I think?

If we can't get OBK into the official repos in the next coming months, then we should go for the Open Build Service as it seems only the viable option for distributing the package through the package managers.

cc @natrys

**Comment by ahmubashshir** [comment]
I can try contacting debian maintainers :3 I submitted a patch on anbox 
once :3

**Reply by mominul** [reply]
That would be great! Do we have to communicate through IRC, mailing lists or they have migrated to something modern like matrix, zulip? 😅 Please cc me on the discussions so I could be informed about what changes would be needed.
Should we contract the ibus-avro maintainer? I have seen him be quite helpful with maintaining ibus-avro.

**Comment by ahmubashshir** [comment]
Last time, I mailed to the maintainer of anbox... but first we need to 
create an RFP[1]


[1]: <https://wiki.debian.org/RFP>

**Comment by ahmubashshir** [comment]
<https://www.debian.org/devel/wnpp/>

**Comment by ahmubashshir** [comment]
should I create an RFP now??

**Reply by bdeshi** [reply]
> should I create an RFP now??

@mominul debian maintainers are usually quite helpful towards new package maintainers. we can create an RFP right now, and collect any packaging concerns from debian people. and then any necessary fixes, along with fixes #183 can be rolled into one commit.

**Reply by ahmubashshir** [reply]
btw... I don't want to maintain any deb distribution :| doing so is significantly tougher than PKGBUILD :(

**Reply by mominul** [reply]
I'll try to fillup an RFP, but I am quite unfamiliar with their processes. :sweat:

**Reply by bdeshi** [reply]
I can take care of that. I was actually planning to submit a request to debs after #183 is fixed.
but I'll be slow.

**Reply by mominul** [reply]
Great! I'll resolve #183 then!

**Reply by bdeshi** [reply]
there's already [a debian intent-to-package ticket for openbangla](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=961541) by @safwanrahman

**Reply by bdeshi** [reply]
@safwanrahman do you have any plans to continue with the itp?

**Reply by mominul** [reply]
I think the package name should be `openbangla-keyboard`, the ITP is filled with the package name `openbangla`. I don't have a strong preference for it though. :grin:

**Reply by ahmubashshir** [reply]
+1 for openbangla-keyboard

**Reply by safwanrahman** [reply]
@bdeshi Yes. I have tried to list it to debian repository and talked with @mominul regarding it as well as talked with some guy in IRC. The debian people seems to be much helpful. I will look into it again this weekend.

**Comment by bdeshi** [comment]
fedora packaging intro is here: https://fedoraproject.org/wiki/Join_the_package_collection_maintainers

**Comment by ahmubashshir** [comment]
No problem for me :3 I use manjaro and build obk myself :3

**Comment by khaalidrafi** [comment]
> https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/
> 
> > _To streamline the productivity of the JFrog Platform we will be sunsetting Bintray (including JCenter), GoCenter, and ChartCenter services on May 1st, 2021. Users of these services will need to migrate to the respective canonical repository_
> 
> possible alternatives:
> 
> * [ ]  selfhosted
> * [ ]  github
> * [ ]  packagecloud
> * [ ]  openSUSE Build Service
> * [ ]  include in official distro repos
> * [ ]  ...

Please include it in Arch Linux and Gentoo official repo.

**Reply by ahmubashshir** [reply]
Ask any TU to include it in community repo.

**Reply by khaalidrafi** [reply]
Need 1 more vote.

**Reply by ahmubashshir** [reply]
I already voted :3

**Reply by khaalidrafi** [reply]
Make a post in all your social accounts asking for support through github star, AUR votes and pkgstats

**Reply by ahmubashshir** [reply]
I don't have fb now :3

**Reply by khaalidrafi** [reply]
I mean OpenBangla's social accounts

**Reply by bdeshi** [reply]
maybe we can wait for organic votes

**Reply by ahmubashshir** [reply]
organic vote 🙃

**Reply by bdeshi** [reply]
🙃

**Reply by khaalidrafi** [reply]
I think there are many people who are enjoying this keyboard but not voting. Asking for votes would not be unfair.

**Reply by bdeshi** [reply]
ehhh, maybe. @mominul @Adyel
by the way: the above is just my personal opinion, I wasn't trying to discourage in any way.

**Reply by khaalidrafi** [reply]
You should also promote it in many places. Even, I was also thinking why I didn't know about this keyboard before.

**Reply by bdeshi** [reply]
@krafi99 @ahmubashshir there, the aur package has 10+ votes now. are any of you already in contact with
 trusted users?

**Reply by khaalidrafi** [reply]
I didn't

**Reply by ahmubashshir** [reply]
I don't know any TUs.

**Reply by bdeshi** [reply]
ok i'll try but,
> 16/02/2021, 19:36:45 unless there's a Dev/TU that uses it and can test it, there's not much chance

**Reply by khaalidrafi** [reply]
https://archlinux.org/people/trusted-users/

**Reply by bdeshi** [reply]
i know.

**Reply by khaalidrafi** [reply]
For @ahmubashshir

**Reply by khaalidrafi** [reply]
> ok i'll try but,
> 
> > 16/02/2021, 19:36:45 unless there's a Dev/TU that uses it and can test it, there's not much chance

Did you contact them? What's the update?

**Reply by bdeshi** [reply]
> Did you contact them? What's the update?

@krafi99 contacted in irc. nothing to mention yet. focusing on fedora for now.
